packages/maskbook/src/extension/background-script/EthereumServices/request.ts

```diff
@@ -26,13 +54,51 @@ export async function request<T extends unknown>(requestArguments: RequestArgume
 export async function requestSend(
     payload: JsonRpcPayload,
     callback: (error: Error | null, response?: JsonRpcResponse) => void,
+    overrides?: SendOverrides,
 ) {
     id += 1
-    await INTERNAL_send(
-        {
-            ...payload,
-            id,
-        },
-        callback,
-    )
+    const { providerType = currentProviderSettings.value } = overrides ?? {}
+    const payload_ = {
```

Comment:
suggest, rename the variable name to `preparePayload`

```suggestion
    const preparePayload = {
```